picard/ui/matchdetailsdialog/__init__.py

```diff
@@ -0,0 +1,386 @@
+        for element in self.filter_checkboxes:
+            if element.isChecked():
+                self.keywords[self.col].append(element.text())
+            else:
+                did_filter = True
+
+        # add decorator to show filtered.
+        if did_filter:
+            it = self.horizontalHeaderItem(self.col)
+            oldHeader = it.text()
+            newHeader = oldHeader + "*"
+            it.setText(newHeader)
+
+        self.filterdata()
+        self.menu.close()
+
+    def clearFilter(self):
+        # clear filter
+        for i in range(self.rowCount()):
+            self.setRowHidden(i, False)
+
+        # remove filter decoroator
+        for i in range(self.columnCount()):
+            it = self.horizontalHeaderItem(i)
+            oldHeader = it.text()
+            newHeader = re.sub('[*+]', '', oldHeader)
+            it.setText(newHeader)
+
+    def filterdata(self):
+
+        columnsShow = dict([(i, True) for i in range(self.rowCount())])
+
+        try:
+            for i in range(self.rowCount()):
+                for j in range(self.columnCount()):
+                    item = self.item(i, j)
+                    if j in self.keywords:
+                        if item.text() not in self.keywords[j]:
+                            columnsShow[i] = False
+            for key, value in columnsShow.items():
+                self.setRowHidden(key, not value)
+
+        except Exception:
+            log.error(traceback.format_exc())
+
+
+class SortableTableWidgetItem(QtWidgets.QTableWidgetItem):
+
+    def __init__(self, sort_key):
+        super().__init__()
+        self.sort_key = sort_key
+
+    def __lt__(self, other):
+        return self.sort_key < other.sort_key
+
+
+class MatchDetailsDialogBase(PicardDialog):
+
+    defaultsize = QtCore.QSize(720, 360)
+    autorestore = False
+    scrolled = pyqtSignal()
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.response_results = []
+        self.setupUi()
+        self.restore_state()
+        # self.columns has to be an ordered dict, with column name as keys, and
+        # matching label as values
+        self.columns = None
+        self.sorting_enabled = True
+        self.create_table()
+        self.finished.connect(self.save_state)
+
+    @property
+    def columns(self):
+        return self.__columns
+
+    @columns.setter
+    def columns(self, list_of_tuples):
+        if not list_of_tuples:
+            list_of_tuples = []
+        self.__columns = OrderedDict(list_of_tuples)
+        self.__colkeys = list(self.columns.keys())
+
+    @property
+    def table_headers(self):
+        return list(self.columns.values())
+
+    def colpos(self, colname):
+        return self.__colkeys.index(colname)
+
+    def set_acoustid_lookup_url_edit(self, p_acoustid_lookup_url):
+        self.acoustid_lookup_url.setText("<a href='" + p_acoustid_lookup_url + "'>" + _("Click to see AcoustId Fingerprint Lookup in browser") + "</a> ")
```

Comment:
Please use formatting instead of string concatenation:

```python
self.acoustid_lookup_url.setText(_('<a href="{url:s}">Click to see AcoustId Fingerprint Lookup in browser</a>').format(url=p_acoustid_lookup_url))
```

It gives context to translators, and they can properly escape string knowing that's html.

Reply:
> please do not rebase each time you make changes it makes those hard to follow, such big set of changes shouldn't be 1 commit.

**Noted going forward.**  Sorry, I thought in the beginning it was suggested that way.  I was "git squashing" it down each time.